server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java

```diff
@@ -116,6 +117,9 @@ public HttpClient get()
 
       httpClient.setIdleTimeout(config.getReadTimeout().getMillis());
       httpClient.setMaxConnectionsPerDestination(config.getNumConnections());
+      final QueuedThreadPool queuedThreadPool = new QueuedThreadPool(config.getNumMaxThreads());
```

Comment:
It looks like this makes non-daemon by default. Suggest setting daemon to true for thread pool